### PR TITLE
Reintroduce Regexp mutator

### PR DESCRIPTION
**Pull request by backus**
* See #565 for initial code review discussion
* See #599 for the revert discussion

### CODE REVIEW

**Comment by backus** [comment]
Includes fix from #596

**Comment by backus** [comment]
@mbj CI is nice and warning free :)

File: lib/mutant/mutator/node/literal/regex.rb [hunk 1]
```diff
@@ -31,6 +32,45 @@ def dispatch
             emit_type(s(:str, NULL_REGEXP_SOURCE), options)
           end
 
+          # Mutate regexp body
+          #
+          # @note will only mutate parts of regexp body if the
+          # body is composed of only strings. Regular expressions
+          # with interpolation are skipped
```
Comment (mbj) [inline lib/mutant/mutator/node/literal/regex.rb]: I think mutant could recurse into the interpolation with ease, and these would also be very interesting, as more likely to reside inside subjects that actually get mutated.

And I'm fine to do this in the next iteration.

Comment (backus) [inline lib/mutant/mutator/node/literal/regex.rb]: Yeah I agree it should be an easy addition

**Comment by backus** [comment]
@mbj build is green now

**Comment by mbj** [comment]
@backus I'm ready to merge after rebase.

**Comment by backus** [comment]
@mbj rebased

**Pull request merged**